### PR TITLE
Consume coredns crd plugin with WRR functionality

### PR DESCRIPTION
**Pull request by ytsarev**

### CODE REVIEW

**Review by k0da — APPROVED** [review]
LGTM

**Review by kuritka — APPROVED** [review]

**Review by kuritka — CHANGES_REQUESTED** [review]
seems terratests fails, requires investigation.

**Comment by ytsarev** [comment]
@kuritka can be related to WRR in action https://github.com/k8gb-io/k8gb/blob/master/terratest/examples/roundrobin_weight1.yaml#L39-L44 ? I assume that before this example was never properly evaluated by terratest with older k8s_crd plugin

**Comment by kuritka** [comment]
@ytsarev , hi, probably. need to check terratests locally.

File: chart/k8gb/templates/coredns-cm.yaml [hunk 1]
```diff
@@ -18,6 +18,7 @@ data:
             resources DNSEndpoint
             filter k8gb.absa.oss/dnstype=local
             negttl {{ .Values.k8gb.dnsZoneNegTTL }}
+            loadbalance weight
```
Comment (k0da) [inline chart/k8gb/templates/coredns-cm.yaml:21]: isn't it hardcoding particular method?

Comment (ytsarev) [reply chart/k8gb/templates/coredns-cm.yaml:19]: Would be great to get confirmation from @kuritka , but if I understand correctly it does standard round-robin if no WRR in Gslb spec, see https://github.com/k8gb-io/coredns-crd-plugin/pull/45

File: chart/k8gb/templates/coredns-cm.yaml [hunk 2]
```diff
@@ -11,13 +11,12 @@ data:
         errors
         health
         ready
-        loadbalance round_robin
         prometheus 0.0.0.0:9153
         forward . /etc/resolv.conf
         k8s_crd {
-            resources DNSEndpoint
```
Comment (ytsarev) [inline chart/k8gb/templates/coredns-cm.yaml:18]: @kuritka it's not not immediately clear from https://github.com/k8gb-io/coredns-crd-plugin/pull/45 - why do we need to remove this config line?

File: chart/k8gb/templates/coredns-cm.yaml (hunk 1 above)
Comment (kuritka) [reply chart/k8gb/templates/coredns-cm.yaml:19]: Hi @ytsarev, RR is only a subset of WRR.  If weights are not defined , the RR is automatically applied.  If weights are defined in CR, the WRR is used. If an error occurs for any reason, then the RR is used automatically. So there should stay `loadbalance weight `. 

The reason why I defines this field is for extendability. I was planning that in the future there will be a `loadbalance consistent` etc.

File: chart/k8gb/templates/coredns-cm.yaml (hunk 2 above)
Comment (kuritka) [reply chart/k8gb/templates/coredns-cm.yaml:18]: `resources DNSEndpoint ` field was meaningless, we didn't work with it in any way.

**Review by kuritka — APPROVED** [review]
lgtm

File: chart/k8gb/templates/coredns-cm.yaml (hunk 2 above)
Comment (ytsarev) [reply chart/k8gb/templates/coredns-cm.yaml:18]: @kuritka cool, I was mostly wondering why it actually breaking the terratest, but cool to get rid of it any case, thanks!

File: chart/k8gb/templates/coredns-cm.yaml (hunk 1 above)
Comment (ytsarev) [reply chart/k8gb/templates/coredns-cm.yaml:19]: Perfect, thank you!

**Pull request merged**